gammapy/background/models.py

```diff
@@ -71,3 +81,775 @@ def evaluate(self, x, y):
         y = np.asanyarray(y, dtype=float)
         parvals = self.parvals(x)
         return self._evaluate_y(y, parvals)
+
+
+def _make_bin_edges_array(lo, hi):
+    """Make bin edges array from a low values and a high values array.
+
+    TODO: move this function to somewhere else? (i.e. utils?)
+
+    Parameters
+    ----------
+    lo : `~numpy.array`
+        lower boundaries
+    hi : `~numpy.array`
+        higher boundaries
+
+    Returns
+    -------
+    bin_edges : `~numpy.array`
+        array of bin edges as [[low], [high]]
+    """
+    return np.append(lo.flatten(), hi.flatten()[-1:])
+
+
+def make_linear_bin_edges_arrays_from_wcs(w, nbins_x, nbins_y):
+    """Make a 2D linear binning from a WCS object.
+
+    This method gives the correct answer only for linear X, Y binning.
+    The method expects angular quantities in the WCS object.
+    X is identified with WCS axis 1, Y is identified with WCS axis 2.
+    The method needs the number of bins as input, since it is not in
+    the WCS object.
+
+    TODO: move this function to somewhere else? (i.e. utils?)
+
+    Parameters
+    ----------
+    w : `~astropy.wcs.WCS`
+        WCS object describing the bin coordinates
+    nbins_x : `~int`
+        number of bins in X coordinate
+    nbins_y : `~int`
+        number of bins in Y coordinate
+
+    Returns
+    -------
+    bins_x : `~numpy.array` of `~astropy.coordinates.Angle`
+        array with the bin edges for the X coordinate
+    bins_y : `~numpy.array` of `~astropy.coordinates.Angle`
+        array with the bin edges for the Y coordinate
+    """
+    # check number of dimensions
+    if w.wcs.naxis != 2:
+        raise ValueError("Expected exactly 2 dimensions, got {}"
+                         .format(w.wcs.naxis))
+
+    unit_x, unit_y = w.wcs.cunit
+    delta_x, delta_y = w.wcs.cdelt
+    delta_x = Angle(delta_x, unit_x)
+    delta_y = Angle(delta_y, unit_y)
+    bins_x = np.arange(nbins_x + 1)*delta_x
+    bins_y = np.arange(nbins_y + 1)*delta_y
+    # translate bins to correct values according to WCS reference
+    # coordinate start empiricaly determined at pix = 0.5: why 0.5?
+    refpix_x, refpix_y = w.wcs.crpix
+    refval_x, refval_y = w.wcs.crval
+    refval_x = Angle(refval_x, unit_x)
+    refval_y = Angle(refval_y, unit_y)
+    bins_x += refval_x - (refpix_x - 0.5)*delta_x
+    bins_y += refval_y - (refpix_y - 0.5)*delta_y
+
+    return bins_x, bins_y
+
+
+def make_linear_wcs_from_bin_edges_arrays(name_x, name_y, bins_x, bins_y):
+    """Make a 2D linear WCS object from arrays of bin edges.
+
+    This method gives the correct answer only for linear X, Y binning.
+    X is identified with WCS axis 1, Y is identified with WCS axis 2.
+
+    TODO: move this function to somewhere else? (i.e. utils?)
```

Comment:
OK, if you want to put this function into a good place now, I think it could go either in
* `gammapy.utils.coordinates`
* `gammapy.utils.wcsutils` (doesn't exist yet, make a new file)
* `gammapy.image.utils`

We'll probably need this again e.g. when we have effective area lookups that are non-radially-symmetric, so maybe it's not bad to move this to a good place now.
It doesn't really have to do anything with background specifically, so in the long-term it won't stay here.

@mapazarr – Your choice!